Realm/RLMAsyncTask.mm

```diff
@@ -127,12 +128,22 @@ - (instancetype)initWithConfiguration:(RLMRealmConfiguration *)configuration
 }
 
 - (void)waitForOpen:(RLMAsyncOpenRealmCallback)completion {
-    {
-        std::lock_guard lock(_mutex);
-        _completion = completion;
-        if (_cancel) {
-            return [self reportError:s_canceledError];
+    __weak auto weakSelf = self;
+    [self waitWithCompletion:^(NSError *error) {
```

Comment:
This is both to sidestep the sendable checking issues with cancellation handlers and to align the APIs for the two tasks so that the Swift code doesn't need to be duplicated. Passing a RLMRealm to the completion handler gives a spurious warning because when it's wrapped in a completion handler there's potentially a hop away from the correct executor and then back, and reading it from the token instead avoids that. These means that Swift wants a completion handler which just returns an error for both task types, but the obj-c (and Swift `Realm.asyncOpen()`) API needs a completion handler with a RLMRealm and error, so this wraps the applicable one to expose that.